ocaml/xapi/xapi_vdi.mli

```diff
@@ -0,0 +1,200 @@
+(*
+ * Copyright (C) 2017 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+(** Checks to see if an operation is valid in this state. Returns Some exception
+    if not and None if everything is ok. *)
```

Comment:
I'm not sure about this, but according to Real World OCaml, this comment is associated with the whole module, I remember reading that ocamldoc comments are tricky.
https://realworldocaml.org/v1/en/html/the-compiler-frontend-parsing-and-type-checking.html